File: src/api/FleetManager/FleetManager.Domain/Vehicles/Inspection.cs
namespace FleetManager.Domain.Vehicles
{
    public record Inspection
    {
        public Guid Id { get; private set; }
        public Guid VehicleId { get; private set; } 
        public DateTime Date { get; private set; } 
        public string Description { get; private set; } 
        public decimal Cost { get; private set; }

        public Inspection(Guid vehicleId, DateTime date, string description, decimal cost)
        {
            Id = Guid.NewGuid();
            VehicleId = vehicleId;
            Date = date;
            Description = description;
            Cost = cost;
        }
    }

}

File: src/client/web/FleetManager/src/app/features/vehicles/vehicle-management/models/inspection.ts
export class Inspection{
  public vehicleId: string;
  public date: Date;
  public description: string;
  public cost: number;

  constructor(vehicleId: string, date: Date, description: string, cost: number) {
    this.vehicleId = vehicleId;
    this.date = date;
    this.description = description;
    this.cost = cost;
  }
}
